File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/tasks/BaseTask.java
package com.tyron.builder.tasks;

import com.tyron.builder.gradle.internal.tasks.VariantAwareTask;
import com.tyron.builder.plugin.builder.AndroidBuilder;

import org.gradle.api.DefaultTask;
import org.gradle.api.tasks.Internal;
import org.gradle.workers.WorkerExecutor;
import org.jetbrains.annotations.NotNull;

import java.io.File;

import javax.inject.Inject;

public abstract class BaseTask extends DefaultTask implements VariantAwareTask {

    @NotNull
    @Override
    public String getVariantName() {
        return "debug";
    }

    @Override
    public void setVariantName(@NotNull String value) {

    }

    @Inject
    public abstract WorkerExecutor getWorkerExecutor();

    @Internal
    public AndroidBuilder getBuilder() {
        throw new UnsupportedOperationException();
    }

    protected static void emptyFolder(File folder) {
        deleteFolder(folder);
        folder.mkdirs();
    }

    protected static void deleteFolder(File folder) {
        File[] files = folder.listFiles();
        if (files != null && files.length > 0) {
            for (File file : files) {
                if (file.isDirectory()) {
                    deleteFolder(file);
                } else {
                    file.delete();
                }
            }
        }
        folder.delete();
    }
}

File: build-tools/codeassist-builder-plugin/src/main/java/com/tyron/builder/tasks/BaseTask.kt
package com.tyron.builder.tasks

import org.gradle.api.DefaultTask
import org.gradle.api.provider.Property
import org.gradle.api.tasks.Internal
import org.gradle.work.DisableCachingByDefault
import org.gradle.workers.WorkerExecutor
import javax.inject.Inject

/**
 * Root Task class for all of AGP.
 *
 * DO NOT EXTEND THIS METHOD DIRECTLY. Instead, extend:
 * - [NewIncrementalTask] -- variant aware task
 * - [NonIncrementalTask] -- variant aware task
 * - [NonIncrementalGlobalTask] -- non variant aware task
 */
@DisableCachingByDefault
abstract class BaseTask : DefaultTask() {
    @get:Internal("only for task execution")
    abstract val projectPath: Property<String>

    @get:Inject
    abstract val workerExecutor: WorkerExecutor

//    @get:Internal
//    abstract val analyticsService: Property<AnalyticsService>

//    /**
//     * Called by subclasses that want to record something.
//     *
//     * The task execution will use [GradleBuildProfileSpan.ExecutionType.TASK_EXECUTION_ALL_PHASES]
//     * as the span type to record the [AndroidVariantTask.recordedTaskAction].
//     */
//    protected inline fun recordTaskAction(
//        analyticsService: AnalyticsService?,
//        crossinline block: () -> Unit
//    ) {
//        Blocks.recordSpan<Exception>(
//            path,
//            GradleBuildProfileSpan.ExecutionType.TASK_EXECUTION_ALL_PHASES,
//            analyticsService
//        ) { block() }
//    }
}